### ISSUE TITLE
Docker's Container on docker hub

### ISSUE THREAD

**Issue by szala1981** [SamR1/FitTrackee#126]
I am a Linux novice and I cannot install fitrackee following the instructions - I have error:

```
root@debian:~/FitTrackee# make docker-build docker-run docker-init
make: docker-compose: No such file or directory
make: *** [Makefile:28: docker-build] Error 127
```

Is it possible to create a container on docker hub? - this is the easiest way to install for many users.
Thanks!

**Comment by SamR1** [comment]
Hi,

Sorry for the delay.

Can you check if [`docker-compose`](https://docs.docker.com/compose/reference/) is correctly installed by displaying its version?

```
$ docker-compose -v
Docker Compose version 2.2.2
```

Note that the docker files present in this repo are just for evaluation.
Another [repository](https://github.com/SamR1/docker-fittrackee) exists with **sample** files for custom configuration.

I can look how to create container on Docker Hub, but I have a basic knowledge of docker since I'm not using it for development or deployment.

**Comment by d-rk** [comment]
@SamR1 are you interested in a PR with a github action workflow to deploy to docker hub automatically. I could help with that.

**Comment by SamR1** [comment]
Hi,

Thanks for your help offer :)
No problem for a PR.

**Comment by albertodiazsaez** [comment]
Hi.

I'm also interested on this because I use FitTrackee on a daily for my bike-commuting, and updating using docker is tedious in it's current state (I understand is for developing purposes and not for "production" use).

How is this doing? I'm not an expert in docker, but I have a bit of experience and if help is needed to create the DockerHub and docker-compose for a  "production-ready" deployment, I think I can help.

**Comment by titilambert** [comment]
@SamR1 Do you have an dockerhub account ? If yes, could you write your username ? thanks !

**Comment by SamR1** [comment]
Hi,

Thanks for the link.

I don't think if it's necessary to create a Javascript image, the Python application already includes [js assets](https://github.com/SamR1/FitTrackee/issues/199#issuecomment-1169808945).

Since I have no experience deploying in Docker images in production, if you or anyone else wants to start a PR, it would be great :).

I have just created a [repository](https://hub.docker.com/r/fittrackee/fittrackee) on Docker Hub.

If you have any questions, you can reach me on [matrix](https://matrix.to/#/#fittrackee:matrix.org). But FYI, at the moment I don't have much free time to work on FitTrackee :/. So, sorry in advance if it takes me a while to reply.

**Comment by DavidHenryThoreau** [comment]
I've read this (https://dockerlabs.collabnix.com/intermediate/workshop/DockerCompose/push_command.html) and make PR

**Comment by DavidHenryThoreau** [comment]
Or even better https://docs.docker.com/docker-hub/builds/link-source/

**Comment by DavidHenryThoreau** [comment]
I've set workflow, after merging into master it generates images as you can see : 
https://github.com/DavidHenryThoreau/FitTrackee/pkgs/container/fittrackee

Commit is here : 
https://github.com/DavidHenryThoreau/FitTrackee/commit/0f58f34dfdb95f3cddd0dbb87d163eff5b0bbe2f?diff=unified&w=0

**Comment by goliath888** [comment]
> I've set workflow, after merging into master it generates images as you can see : https://github.com/DavidHenryThoreau/FitTrackee/pkgs/container/fittrackee
> 
> Commit is here : [DavidHenryThoreau@0f58f34?diff=unified&w=0](https://github.com/DavidHenryThoreau/FitTrackee/commit/0f58f34dfdb95f3cddd0dbb87d163eff5b0bbe2f?diff=unified&w=0)

So, it is possible to use the ghcr.io/davidhenrythoreau/fittrackee:sha-971cf6a image? 
But how we should configure it in order to run it? I mean, which ports, env variables, etc should we configure? 
The same as the docker-compose-dev.yml file?

**Comment by DavidHenryThoreau** [comment]
> > I've set workflow, after merging into master it generates images as you can see : https://github.com/DavidHenryThoreau/FitTrackee/pkgs/container/fittrackee
> > Commit is here : [DavidHenryThoreau@0f58f34?diff=unified&w=0](https://github.com/DavidHenryThoreau/FitTrackee/commit/0f58f34dfdb95f3cddd0dbb87d163eff5b0bbe2f?diff=unified&w=0)
> 
> So, it is possible to use the ghcr.io/davidhenrythoreau/fittrackee:sha-971cf6a image? But how we should configure it in order to run it? I mean, which ports, env variables, etc should we configure? The same as the docker-compose-dev.yml file?

You can use that [`docker-compose.yaml`](https://github.com/DavidHenryThoreau/FitTrackee/blob/master/docker-compose.yaml) file

**Comment by SamR1** [comment]
Docker images are now available on [Docker Hub](https://hub.docker.com/r/fittrackee/fittrackee/tags) and [GitHub container registry](https://github.com/SamR1/FitTrackee/pkgs/container/fittrackee). 

Link to documentation: https://samr1.github.io/FitTrackee/en/installation.html#id6

Thank to @DavidHenryThoreau for initiating the pull request.

**Comment by SamR1** [comment]
released in v0.8.13
